docs/compatibility.md

```diff
@@ -83,6 +83,20 @@ after Spark 3.1.0.
 We do not disable operations that produce different results due to `-0.0` in the data because it is
 considered to be a rare occurrence.
 
+### `NaN` vs `NaN`
+
+Apache Spark does not have a consistent way to handle `NaN` comparison. Sometimes, All `NaN` are
+considered as one unique value while other times they can be treated as different. The outcome of
+`NaN` comparision can differ in various operations and also changes over time. Due to such complexity,
+our plugin cannot guarantee to always match its output with Apache Spark if there are `NaN` values
+in the input.
```

Comment:
I agree we need to be very specific and Ideally point to specific issues that we have to fix the problems, even if they are low priority because we don't think NaN is common in these specific cases.